### PR TITLE
{chem}[GCCcore/9.3.0] XCFun 2.1.0

### PR DESCRIPTION
**Pull request by stigrj**
There exists an old version under the name `XCfun`, but the name really should be [XCFun](https://xcfun.readthedocs.io/en/latest/index.html), so where do you want me to put this, @boegel?

### CODE REVIEW

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (Micket) [inline easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:16]:
```suggestion
    ('CMake', '3.16.4')
```

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (Micket) [inline easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:18]: No more python dep?

**Review by Micket — CHANGES_REQUESTED** [review]
So, the name question; Since it's not a widely used dependency, i suppose we can consider renaming it. If we do, we need to update the dependency name in PySCF as well.

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (stigrj) [reply easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:18]: Well, XCFun is a C++ library with an optional Python interface, and this particular build would be without Python support. I tried first to make the general `XCFun-2.1.0-foss-2020a-Python-3.8.2.eb` version, but it pulled in some more dependencies which I don't really need (but PySCF does). Do you think it's worth having a separate Python-free version like this, or should we only provide the general version _with_ Python support?

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node101.hydra.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/c65e6bd6c209af646dccb525e38b6621 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node369.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/d3fed647f0fe654949ea854220bfa1f0 for a full test report.

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (Micket) [reply easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:18]: Typically I would say, do opt in, unless there is some reason not to. 

If we don't want to opt it, then it might be possible to move this up the hierarchy.

Comment (stigrj) [reply easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:18]: Not sure what you mean with moving up the hierarchy?

I discussed briefly with XCFun maintainer @robertodr. The Python interface is not widely used, so we think a separate recipe without Python support is worth while.

Comment (lexming) [reply easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:18]: I agree that we can merge this easyconfig without Python as the name of the easyconfig will be different in any case. If anybody needs support for Python it can be added later on.

@stigrj  @Micket was talking about changing the toolchain of the easyconfig. Since XCFun does not use BLAS/LAPACK, FFTW or MPI, then it does not need the `foss` toolchain. It can be moved to `GCC-10.2.0`, making it more compatible with other easyconfigs.

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (lexming) [inline easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb:6]:
```suggestion
https://xcfun.readthedocs.io
```

**Review by lexming — CHANGES_REQUESTED** [review]
As discussed, I think that this can be merged once it is moved to `GCC` and the homepage URL is fixed (see below)

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (stigrj) [reply easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb]: Thanks for the input @lexming! I switched to `GCCcore-9.3.0`, I don't think 10 is available yet on my clusters :slightly_smiling_face:

File: easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb
Comment (stigrj) [reply easybuild/easyconfigs/x/XCFun/XCFun-2.1.0-foss-2020a.eb]: done

**Comment by boegel** [comment]
Can we make sure an accompanying (separate) PR is opened to rename the existing `XCfun` to `XCFun` (which is a bugfix) when this is merged?

**Comment by stigrj** [comment]
See #11603

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/923fd50182968f26d32ba26c6ce132e1 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node101.hydra.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/5ea22375b4ce0e0d303917a50a1caefe for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @stigrj !

**Pull request merged**
